src/event/format/json.rs

```diff
@@ -52,8 +52,17 @@ impl EventFormat for Event {
         static_schema_flag: Option<&String>,
         time_partition: Option<&String>,
         schema_version: SchemaVersion,
+        log_source: &str,
```

Comment:
We should move away from using str given how wide the possibilities are, please consider using enums similar to what we are doing with `SchemaVersion` above, it will also make things easier to read